src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -520,6 +540,14 @@ protected override AuthenticationTicket AuthenticateCore()
             }
         }
 
+        protected virtual IdentityModel.Clients.ActiveDirectory.AuthenticationResult RedeemAuthorizationCode(string authorizationCode)
```

Comment:
@PinpointTownes We want to use ADAL for redeeming code for tokens instead of doing a direct httpClient call. That is why we have adal for :smile: We have plans to extend this to authorization servers beyond AAD but that will be the second step.